nikolay_shapurov/graphprinter.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef GRAPHPRINTER_H
+#define GRAPHPRINTER_H
+
+#include <string>
+
+class Graph;
+
+class GraphPrinter {
+ public:
+  GraphPrinter(const Graph& graph) : graph_(graph) {}
+  std::string print() const;
+
+ private:
+  std::string print_vertices() const;
+  std::string print_edges() const;
```

Comment:
В плане интерфейса, разумней будет иметь методы по распечатке всех сущностей, вместо их массива:
```suggestion
  std::string print_vertex(const Vertex& vertex) const;
  std::string print_edge(const Edge& edge) const;
```
Такой интерфейс будет более гибким. И более того, их можно вынести в `public` и тогда можно будет использовать как отдельные методы, при необходимости.